projects/fsastorefrontlib/src/cms-components/user-profile/user-profile.component.html

```diff
@@ -0,0 +1,212 @@
+                  >{{
+                    'fscommon.register.dateOfBirth.label' | cxTranslate
+                  }}:</span
+                >
+                <span class="notice">
+                  {{ user.dateOfBirth | parseDate | date: 'dd MMM yyyy' }}
+                </span>
+              </li>
+            </ul>
+          </div>
+        </div>
+      </div>
+    </div>
+    <ng-container
+      *ngIf="{
+        quotes: customerQuotes$ | async,
+        policies: customerPolicies$ | async,
+        claims: customerClaims$ | async
+      } as customerAssets"
+    >
+      <div class="col-md-7">
+        <div class="box-shadow border border-color-3">
+          <h5 class="text-center py-3 mb-0">
+            {{ 'dashboard.overview' | cxTranslate }}
+          </h5>
+          <div class="dashboard-overview-wrapper background-color-4">
+            <ul
+              class="dashboard-overview-stats mb-0 bg-white border border-color-3 p-3"
+            >
+              <li class="item text-break text-center text-uppercase">
+                <div class="number notice semi-bold">
+                  {{
+                    customerAssets.claims?.claims
+                      ? customerAssets.claims?.claims?.length
+                      : '0'
+                  }}
+                </div>
+                <span class="description">{{
+                  'claim.claims' | cxTranslate
+                }}</span>
+              </li>
+              <li class="item text-break text-center text-uppercase">
+                <div class="number notice semi-bold">
+                  {{
+                    customerAssets.policies?.insurancePolicies
+                      ? customerAssets.policies?.insurancePolicies?.length
+                      : '0'
+                  }}
+                </div>
+                <span class="description">{{
+                  'policy.policies' | cxTranslate
+                }}</span>
+              </li>
+              <li class="item text-break text-center text-uppercase">
+                <div class="number notice semi-bold">
+                  {{
+                    customerAssets.quotes?.insuranceQuotes
+                      ? customerAssets.quotes?.insuranceQuotes?.length
+                      : '0'
+                  }}
+                </div>
+                <span class="description">
+                  {{ 'fscommon.quotesAndApplications' | cxTranslate }}
+                </span>
+              </li>
+            </ul>
+            <div class="dashboard-overview-assets mt-3">
+              <div
+                *ngFor="let asset of customerAssets | keyvalue; let i = index"
+                [ngClass]="assetSelected === asset.key ? 'active' : ''"
+                class="asset bg-white p-2 align-items-center"
+                role="button"
+                [tabindex]="i + 1"
+                (click)="
+                  showAssetList(
+                    asset.key === 'quotes'
```

Comment:
Same here: Didn't we say it is better to remove complex logic from template to component.ts file?

Reply:
Yes of course, but here we are checking only one property against strings and passing the resulting object. If we are to do it in the component, we would have to use subscription there and here in the template and probably use `shareReplay()` again. This way we unpack everything here and just checking one property. Also, since this approach is not generic at all, as stated above, we have a follow up task that should use configuration to resolve this anyway. So there shouldn't be any checking like this done anywhere, since I agree this is cumbersome.